From ec063b26bd632d5e8822546e3c654edb656c9045 Mon Sep 17 00:00:00 2001
From: Bai Ping <b51503@freescale.com>
Date: Wed, 19 Aug 2015 22:02:58 +0800
Subject: [PATCH 0246/5242] MLK-11389-01 ARM: imx: keep M/F power domain on
 after suspend

commit  191b7b40b9e8f536d567b047435faf6fe4bb30dc from
https://source.codeaurora.org/external/imx/linux-imx.git

After the system resume back from suspend, the M/F mix domain
power down bit should be cleared to make sure this domain's power
is on in other low power mode.

Signed-off-by: Bai Ping <b51503@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/gpc.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-imx/gpc.c b/arch/arm/mach-imx/gpc.c
index ed779c9..d5f11c4 100644
--- a/arch/arm/mach-imx/gpc.c
+++ b/arch/arm/mach-imx/gpc.c
@@ -99,6 +99,9 @@ void imx_gpc_post_resume(void)
 
 	/* Keep ARM core powered on for other low-power modes */
 	imx_gpc_set_arm_power_in_lpm(false);
+	/* Keep M/F mix powered on for other low-power modes */
+	if (cpu_is_imx6ul())
+		writel_relaxed(0x0, gpc_base + GPC_PGC_MF_PDN);
 
 	for (i = 0; i < IMR_NUM; i++)
 		writel_relaxed(gpc_saved_imrs[i], reg_imr1 + i * 4);
-- 
1.7.9.5

